cmake/dependencies.cmake

```diff
@@ -148,7 +148,12 @@ if(THREADING STREQUAL "TBB" OR THREADING STREQUAL "TBB_AUTO")
     endif()
 
     update_deps_cache(TBBROOT "${TBB}" "Path to TBB root folder")
-    update_deps_cache(TBB_DIR "${TBB}/cmake" "Path to TBB cmake folder")
+    if(EXISTS "${TBBROOT}/lib/cmake/TBB/TBBConfig.cmake")
```

Comment:
the path is ${TBBROOT}/lib/cmake/**tbb**/TBBConfig.cmake
this makes no diff on Windows, but fails on linux (whihc is case-senstitive for path)

Reply:
From oneTBB side we can consider aligning `tbb` and `TBB` in the path for different distributions, but you still may get an issue with `lib` part. We use `CMAKE_INSTALL_LIBDIR` from [GNUInstallDirs](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html). It may be `lib` or `lib64` or `lib/<multiarch-tuple>`.

So I'd propose to consider the following way of finding `TBBConfig.cmake`:
```cmake
file(GLOB_RECURSE TBB_CONFIG_FILE ${TBBROOT}/*/TBBConfig.cmake)
get_filename_component(TBB_DIR ${TBB_CONFIG_FILE} DIRECTORY)
```

BTW, adding `${TBBROOT}` path into `CMAKE_PREFIX_PATH` variable should also make `find_package(TBB)` working. As alternative to `TBB_DIR`.